### ISSUE TITLE
WASM: support VirtualizingPanel in ComboBox/CarouselPanel

### ISSUE THREAD

**Issue by yowl** [nventive/Uno#556]
## I'm submitting a...

 - Feature request 

## Current behavior
The ComboBox does not virtualize the items leading to poor performance

## Expected behavior
UWP uses a CarouselPanel which extends from VirtualizingPanel to give good performance, wasm should do the same.

## Minimal reproduction of the problem with instructions
Add a combo with a lot of items and try to open it in wasm, its slow even given its interpreted.

## Environment

```
Nuget Package: 
    <PackageReference Include="Uno.UI" Version="1.43.0-dev.746" />
    <PackageReference Include="Uno.Wasm.Bootstrap" Version="1.0.0-dev.159" />
Package Version(s): 

Affected platform(s):
- [ ] iOS
- [ ] Android
- [ x] WebAssembly
- [ ] Windows
- [ ] Build tasks

Visual Studio
- [x ] 2017 (version: )
- [ ] 2017 Preview (version: )
- [ ] for Mac (version: )

Relevant plugins
- [ ] Resharper (version: )
```

**Comment by davidjohnoliver** [comment]
Implemented for WASM by #1134.

**Comment by pkar70** [comment]
Is there any hope for quick ComboBox? Without 25 seconds delay for adding ~200 items?

**Comment by davidjohnoliver** [comment]
You can try the following:

```xaml
			<not_win:ComboBox.ItemsPanel>
				<ItemsPanelTemplate>
					<ItemsStackPanel />
				</ItemsPanelTemplate>
			</not_win:ComboBox.ItemsPanel>
```

**Comment by pkar70** [comment]
Current XAML:

`    <ComboBox x:Uid="resAll" Name="uiStopList" Grid.Row="2" Header="Tram" Margin="0,20,0,15" DoubleTapped="uiStopList_DoubleTapped" SelectionChanged="uiStopList_SelectionChanged"/>`

Convert it to:
`            <ComboBox x:Uid="resAll" Name="uiStopList" Grid.Row="2" Header="Tram" Margin="0,20,0,15" DoubleTapped="uiStopList_DoubleTapped" SelectionChanged="uiStopList_SelectionChanged">`
`			<not_win:ComboBox.ItemsPanel>`
`				<ItemsPanelTemplate>`
`					<ItemsStackPanel />`
`				</ItemsPanelTemplate>`
`			</not_win:ComboBox.ItemsPanel>`
`</ComboBox>`

?

**Comment by jeromelaban** [comment]
@davidjohnoliver did we have some update there, can changing the panel improve the situation ?

**Comment by davidjohnoliver** [comment]
One thing to consider would be using the 'managed' ListView implementation for the ComboBox ItemsPanel, since it works on Android and iOS, and many of the features it lacks (grid layout, inline header/footer, grouping etc) aren't applicable to ComboBox anyway.

**Comment by davidjohnoliver** [comment]
@yowl No, it should be fine. :) The approach I floated would only be used for the default ComboBox behaviour, ie when ItemsPanel is a CarouselPanel. If you're overriding the ItemsPanel then you won't be affected. (By 'grid layout' I meant the ItemsWrapGrid panel, that wasn't very clear.)

**Comment by MartinZikmund** [comment]
From #8897 -

> When navigating to a page containing several combobox with a lot of items (~250), it takes a lot of time to load the page. In the example above, I am showing this is the case even when the combobox is collapsed.

This is important mention here - the performance degradation applies even if the combo is never opened.

**Comment by yowl** [comment]
Has it been fixed?

**Comment by jeromelaban** [comment]
Indeed, yes, it's fixed in the Skia Render targets for all platforms.
